lint/file.go

```diff
@@ -96,7 +96,7 @@ func (f *File) isMain() bool {
 
 const directiveSpecifyDisableReason = "specify-disable-reason"
 
-func (f *File) lint(rules []Rule, config Config, failures chan Failure) {
+func (f *File) lint(rules []Rule, config Config, failures chan Failure) error {
```

Comment:
Once this method returns an error, what to do with it? (as is, the caller just ignores the error)

Reply:
that is good point what next, we should handle that here:

```pkg.lint(ruleSet, config, failures)``` and return error and it should finish in:

```
if err := l.lintPackage(pkg, gover, ruleSet, config, failures); err != nil {                                                                                   
  fmt.Fprintln(os.Stderr, err)                                                                                                                               
  os.Exit(1)                                                                                                                                                 
 }
 ```

Reply:
Do we exit at `pkg.lint` or should we propagate the error up to the main function and exit there?

Reply:
we should handle pkg.ling err and return because level up is:

```
if err := l.lintPackage(pkg, gover, ruleSet, config, failures); err != nil {                                                                                   
  fmt.Fprintln(os.Stderr, err)                                                                                                                               
  os.Exit(1)                                                                                                                                                 
 }
 ```

Reply:
I agree with this change. Panic should be avoided